src/rules/selector-no-redundant-nesting-selector/index.js

```diff
@@ -1,27 +1,34 @@
 import { utils } from "stylelint";
-import { namespace, parseSelector } from "../../utils";
+import {
+  namespace,
+  parseSelector,
+  hasNestedSibling,
+  isKeyword
+} from "../../utils";
 
 export const ruleName = namespace("selector-no-redundant-nesting-selector");
 
 export const messages = utils.ruleMessages(ruleName, {
   rejected: "Unnecessary nesting selector (&)"
 });
 
-export default function(actual) {
+export default function(actual, options) {
   return (root, result) => {
     const validOptions = utils.validateOptions(result, ruleName, { actual });
```

Comment:
Oh sorry I totally missed it on the first review round, but the `utils.validateOptions` call should include rule's secondary options too.

Have a look at how it's done here:
https://github.com/stylelint/stylelint/blob/master/lib/rules/font-family-no-duplicate-names/index.js#L25-L36

The option could also accept a regex as a value like the rule I just linked.